File: src/app/interfaces/order.ts
export interface Order {
  status: OrderStatus;
  totalPrice: number;
  deliveryTime: string;
  comment: string;
  paymentId: string | null;
  paymentMethod: string;
  orderMenus: OrderMenu[];
  userId: number;
}

export enum OrderStatus {
  niezapłacone = 'niezapłacone',
  zapłacone = 'zapłacone',
}

export interface OrderMenu {
  menuId: number;
  quantity: number;
}

export interface OrderResponse {
  message: string;
  orderId: string;
}

export interface OrderDTO {
  id: number;
  deliveryTime: string;
  orderMenus: OrderMenu[];
  paymentId: string | null;
  paymentMethod: string;
  comment: string;
  status: OrderStatus;
  totalPrice: number;
  orderDate: string;
}

export interface AdminOrderDTO {
  id: number;
  status: OrderStatus;
  orderDate: string;
  deliveryTime: string;
  paymentMethod: string;
  paymentId: string;
  comment: string;
  totalPrice: number;
  orderMenus: OrderMenu[];
  userName: string;
  userEmail: string;
  userPhone: string;
}

File: backend/src/main/java/com/urban/backend/model/Order.java
package com.urban.backend.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import com.urban.backend.enums.OrderStatus;

import java.time.ZoneId;
import java.time.ZonedDateTime;
import java.util.List;

@Entity
@Setter
@Getter
@Table(name = "orders", schema = "taw_restaurant")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status;

    @Column(nullable = false)
    private double totalPrice;

    @Column(nullable = false)
    private String deliveryTime;

    private String comment;

    private String paymentId;

    @Column(nullable = false)
    private String paymentMethod;

    private ZonedDateTime orderDate;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;


    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<OrderMenu> orderMenus;

    @PrePersist
    protected void onCreate() {
        orderDate = ZonedDateTime.now(ZoneId.of("Europe/Warsaw"));
    }
}
